Source/Core/VideoCommon/FPSCounter.cpp

```diff
@@ -44,22 +45,18 @@ void FPSCounter::LogRenderTimeToFile(u64 val)
 
 void FPSCounter::Update()
 {
-  const u64 time = Common::Timer::NowUs();
-  const u64 diff = time - m_last_time;
-  m_time_diff_secs = static_cast<double>(diff / 1000000.0);
+  const s64 time = Common::Timer::NowUs();
+  const s64 diff = std::max<s64>(1, time - m_last_time);
+  m_last_time = time;
+
   if (g_ActiveConfig.bLogRenderTimeToFile)
     LogRenderTimeToFile(diff);
 
-  m_frame_counter++;
-  m_time_since_update += diff;
-  m_last_time = time;
+  m_raw_dt = static_cast<double>(diff / 1000000.0);
 
-  if (m_time_since_update >= FPS_REFRESH_INTERVAL_US)
-  {
-    m_fps = m_frame_counter / (m_time_since_update / 1000000.0);
-    m_frame_counter = 0;
-    m_time_since_update = 0;
-  }
+  const double raw_fps = 1.0 / m_raw_dt;
+  const double a = 1.0 - std::exp(-m_raw_dt / FPS_COUNTER_RC);
```

Comment:
This is fairly understandable so it's not a big deal, but is there any name we could use to better represent what this actually does?

Reply:
I'm not exactly sure of a name other than `a` as thats what is what is used by most math papers.

Reply:
I'm not sure what's the procedure on dolphin for external documentation, but a reference to the algorithm would be beneficial IMO, to avoid people having to look up the git history.

Reply:
It's a little awkward though because I kind of want to phase out the usage of this algorithm for the one in the next PR which I view as objectively better.